rllib/execution/parallel_requests.py

```diff
@@ -182,3 +183,224 @@ def wait_asynchronous_requests(
+        num_requests_launched = 0
+        available_actor = self._get_available_actor()
+        while available_actor and self._call_queue.qsize() > 0:
+            remote_fn, fn_args, fn_kwargs = self._call_queue.get()
+            req = available_actor.apply.remote(remote_fn, *fn_args, **fn_kwargs)
+            self._remote_requests_in_flight[available_actor].add(req)
+            self._pending_remotes.append(req)
+            self._pending_to_actor[req] = available_actor
+            available_actor = self._get_available_actor()
+            num_requests_launched += 1
+        return num_requests_launched
+
+    def get_ready_requests(self) -> Dict[ActorHandle, List[Any]]:
+        """Get requests that are ready to be returned
+
+        Returns:
+            A dictionary of actor handles to lists of returns from tasks that were
+             previously submitted to this actor pool that are now ready to be returned.
+             If return_object_refs
+
+        """
+        ready_requests_dict = defaultdict(list)
+        ready_requests, self._pending_remotes = ray.wait(
+            self._pending_remotes,
+            timeout=self._ray_wait_timeout_s,
+            num_returns=len(self._pending_remotes),
+        )
+        if not self._return_object_refs:
+            objs = ray.get(ready_requests)
+        else:
+            objs = ready_requests
+        for req, obj in zip(ready_requests, objs):
+            actor = self._pending_to_actor[req]
+            self._remote_requests_in_flight[actor].remove(req)
+            ready_requests_dict[actor].append(obj)
+            if actor in self._unavailable_workers:
+                self._available_workers.append(actor)
+                self._available_workers_set.add(actor)
+                self._unavailable_workers.remove(actor)
+            del self._pending_to_actor[req]
+        del ready_requests
+        self._run()
+        return dict(ready_requests_dict)
+
+    def _get_available_actor(self) -> Optional[ActorHandle]:
+        """Get an available actor to run a remote request"""
+        available_actor = (
+            self._available_workers[0] if self._available_workers else None
+        )
+        while available_actor:
+            if (
+                len(self._remote_requests_in_flight[available_actor])
+                >= self._max_remote_requests_in_flight
+            ):
+                self._unavailable_workers.add(available_actor)
+                self._available_workers_set.remove(available_actor)
+                self._available_workers.pop(0)
+                available_actor = (
+                    self._available_workers[0] if self._available_workers else None
+                )
+            else:
+                available_actor = self._available_workers.pop(0)
+                self._available_workers.append(available_actor)
+                return available_actor
+        return None
+
+    def add_worker(self, new_worker: ActorHandle) -> None:
```

Comment:
where do we use add_worker() and remove_worker()?
if they are only used in unit tests, they should be part of the public API.